install/kubernetes/cilium/templates/hubble/metrics-service.yaml

```diff
@@ -1,4 +1,27 @@
 {{- if and .Values.hubble.enabled .Values.hubble.metrics.enabled }}
+{{- if .Values.hubble.metrics.serviceMonitor.enabled }}
+apiVersion: v1
+kind: Service
+metadata:
+  name: hubble-metrics
+  namespace: {{ .Release.Namespace }}
+  labels:
+    k8s-app: hubble
+  annotations:
+    {{- with .Values.hubble.metrics.serviceAnnotations }}
+    {{- toYaml . | nindent 4 }}
+    {{- end }}
+spec:
+  clusterIP: None
+  type: ClusterIP
+  ports:
+  - name: hubble-metrics
+    port: {{ .Values.hubble.metrics.port }}
+    protocol: TCP
+    targetPort: hubble-metrics
+  selector:
+    k8s-app: cilium
+{{- else if .Values.hubble.metrics.enabled }}
```

Comment:
this check seems redudant, as we have similar check at line 1.

Reply:
I took the approach from https://github.com/cilium/cilium/blob/master/install/kubernetes/cilium/templates/cilium-agent/service.yaml#L24 so that the `Service` manifests would be templated in the same way.

Your patch would work as well, had to check if applying a manifest with empty `annotations` block works, and it looks to be OK. I'll switch to that approach.

Reply:
Had to add `not` to the if clause so that the annotations are only added when ServiceMonitor is disabled.